backend/ssh.pm

```diff
@@ -0,0 +1,143 @@
+# Copyright © 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package backend::ssh;
+use strict;
+use base 'backend::baseclass';
```

Comment:
As discussed verbally, I think it is nice to use the Mojo base classes, they provide some nice setting and getter functions amongst other things (see OpenQA/Qemu/*). However this class must derive from backend::basclass so it may be a good idea to move some of the functionality here to the library OpenQA/ssh/SomeClass.pm where you can derive from the Mojo base class.